job_executor/adapter/local_storage.py

```diff
@@ -381,4 +382,27 @@ def archive_draft_version(version: str):
     )
 
     if archive_dir.exists():
-        shutil.copyfile(DRAFT_VERSION_PATH, archived_draft_version_path)
\ No newline at end of file
+        shutil.copyfile(DRAFT_VERSION_PATH, archived_draft_version_path)
+
+def archive_input_files(dataset_name: str, json_file: str, csv_file: str = None):
```

Comment:
not sure what ends up looking cleanest here, but I since we know that files in the input directory must be structured like:
```
/DATASET_NAME
    DATASET_NAME.json
    DATASET_NAME.csv # if ADD or CHANGE
```
We can pass this function the dataset_name:str alone, and it copies the entire directory.
```py
    archive_dir = Path(f'{INPUT_DIR}/archive/{dataset_name}')
    timestamp = 123 # generate some timestamp to avoid collisions?
    while archive_dir.exists():
        archive_dir = Path(f'{INPUT_DIR}/archive/{dataset_name}_{timestamp}')
    shutil.copytree(f'{INPUT_DIR}/{dataset_name}', archive_dir)
```

Reply:
I suppose that:
`    archive_dir = Path(f'{INPUT_DIR}/archive/{dataset_name}')
    os.makedirs(archive_dir, exist_ok = True)
    shutil.copytree(f'{INPUT_DIR}/{dataset_name}', archive_dir, dirs_exist_ok=True)
`
is even simpler since it should be ok to unconditionally copy the files since they are immutable and thus no need for a timestamp ?

Reply:
Done

Reply:
This only becomes a problem if a data provider f.ex: imports a dataset 'INNTEKT'. It then gets archived as expected. Then the data provider has a new update for INNTEKT in the next version, and now the last 'INNTEKT' is still archived, which may cause collisions. Not sure how we want to avoid the collision here though 🤔 Thoughts?

Reply:
On second thought I don't see the harm in overwriting it like the code in this PR does now. We don't need to archive several versions back. 👍🏻 Resolved